cfme/infrastructure/virtual_machines.py

```diff
@@ -1518,3 +1527,40 @@ def VIEW(self):     # noqa
 
     def step(self, *args, **kwargs):
         self.prerequisite_view.toolbar.monitoring.item_select('Utilization')
+
+
+@navigator.register(InfraVm, 'PolicySimulation')
+class PolicySimulation(CFMENavigateStep):
+    VIEW = PolicySimulationView
+    prerequisite = NavigateToSibling('Details')
+
+    def step(self, *args, **kwargs):
+        self.prerequisite_view.toolbar.policy.item_select('Policy Simulation')
+
+
+@navigator.register(InfraVmCollection, "PolicySimulation")  # noqa
+class PolicySimulation(CFMENavigateStep):  # noqa
+    VIEW = PolicySimulationView
+
+    @property
+    def navigatable(self):
+        return bool(self.obj.filters.get("name") or self.obj.filters.get("names"))
+
+    def prerequisite(self):
+        provider = self.obj.filters.get("provider")  # None if no filter
+        if provider:
+            return navigate_to(provider, "ProviderVms")
+        else:
+            return navigate_to(self.obj, "All")
+
+    def step(self, *args, **kwargs):
+        # if filter is not defined we don't want to navigate using every object
+        if not self.navigatable:
+            raise DestinationNotFound(
+                "Navigation to policy simulation page using a collection requires the collection "
+                "to be filtered."
+            )
+        # click the checkbox of every object in the filtered collection
+        for entity in self.obj.all():
+            self.prerequisite_view.entities.get_entity(name=entity.name, surf_pages=True).check()
```

Comment:
How does this behave if it checks an item on the first page, then surfs to the 3rd page and selects a 2nd item?

Reply:
Good question, in MIQ if you perform this manually then you're initial check is not preserved. However, in this case, I have seen that the `all()` method of the collection will display 1000 items per page so that everything is displayed. So surfing the page is not relevant here. When the `all()` method is updated to use rest or a DB query, we will need to account for this behavior.